recipes/gpmdp.lua

```diff
@@ -0,0 +1,103 @@
+
+--[[
+
+        Licensed under GNU General Public License v2
+        * (c) 2017, Greg Flynn
+
+--]]
+local awful = require("awful")
+local naughty = require("naughty")
+local io, next, os, string, table = io, next, os, string, table
+
+-- Google Play Music Desktop Player widget
+-- requires: curl and dkjson or lain
+
+local gpmdp = {
+    notify        = "on",
+    followtag     = false,
+    file_location = os.getenv("HOME") .. "/.config/Google Play Music Desktop Player/json_store/playback.json",
+    notification_preset = {
+        title     = "Now playing",
+        icon_size = dpi(128),
+        timeout   = 6
+    },
+    notification  = nil,
+    current_track = nil
+}
+
+function gpmdp.notification_on()
+    local gpm_now = gpmdp.latest
+    gpmdp.current_track = gpm_now.title
+
+    if gpmdp.followtag then gpmdp.notification_preset.screen = awful.screen.focused() end
+    awful.spawn.easy_async(string.format("curl %s -o /tmp/gpmcover.png", gpm_now.cover_url), function(stdout)
```

Comment:
Also, how do we know that this is a PNG? Could other file formats also be possible?